src/dmd/mtype.d

```diff
@@ -481,7 +380,60 @@ alias MOD = ubyte;
  */
 extern (C++) abstract class Type : RootObject
 {
-    TY ty;
+    enum Kind : ubyte
+    {
+        array,     // slice array, aka T[]
+        sarray,    // static array, aka T[dimension]
+        aarray,    // associative array, aka T[type]
```

Comment:
If it needs a comment, it's probably a bad name.  `staticArray` and `associativeArray` don't seem all that bad.